MapboxCoreNavigation/CLLocation.swift

```diff
@@ -114,17 +114,26 @@ extension CLLocation {
             return stepCoordinates
         }
         
-        return nearbyCoordinates
+        return legProgress.nearbyCoordinates
     }
     
+    func projectedDistance(for distanceRemaining: CLLocationDistance) -> CLLocationDistance {
+        // If the user is near the end of the route, do not project the puck forward.
+        // Otherwise, this can cause strange wrapping at maneuvers.
+        guard distanceRemaining > RouteControllerManeuverZoneRadius else {
+            return 0
```

Comment:
this is looking pretty jagged when it occurs on a straight maneuver. I wonder if we can apply some sort of curve to this value that eases in and out nicely.